llvm/include/llvm/Support/VirtualOutputConfig.def

```diff
@@ -0,0 +1,23 @@
+//===- VirtualOutputConfig.def - Virtual output config defs -----*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef HANDLE_OUTPUT_CONFIG_FLAG
+#error "Missing macro definition of HANDLE_OUTPUT_CONFIG_FLAG"
+#endif
+
+// Define HANDLE_OUTPUT_CONFIG_FLAG before including.
+//
+// #define HANDLE_OUTPUT_CONFIG_FLAG(NAME, DEFAULT)
+
+HANDLE_OUTPUT_CONFIG_FLAG(Text, false) // OF_Text.
+HANDLE_OUTPUT_CONFIG_FLAG(CRLF, false) // OF_CRLF.
+HANDLE_OUTPUT_CONFIG_FLAG(CrashCleanup, true) // E.g., RemoveFileOnSignal.
```

Comment:
This is the same as the outputbackend it's replacing, FTR, but I agree it's not a good name. The motivation is to name the intent, rather than the action. RemoveFileOnSignal or RemoveOnSignal feels like it's asking specifically for `sys:: RemoveFileOnSignal()` to be called, which may not make sense for some backends.

How about `DiscardOnSignal`? Matches the name of the OutputFile API (intent) while forming a connection to the familiar sys API (RemoveFileOnSignal). We could even implement it generically in `OutputFile` itself, if we wanted to, eventually.

The other "odd" naming thing is renaming CompilerInstance::createOutputFile()'s UseTemporary parameter to AtomicWrite. I think this name is pretty good, but maybe it'd be better to stick with the familiar. WDYT?

Reply:
(let me know if you're happy enough with that)